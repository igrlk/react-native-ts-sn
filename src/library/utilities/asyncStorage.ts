import { AsyncStorage } from 'react-native';

export const prefix = 'EventNetwork/';

export default class Storage {
	static async getItem(name: string) {
		const item = await AsyncStorage.getItem(prefix + name);

		if (item) return JSON.parse(item);

		return null;
	}

	static setItem(name: string, value: any) {
		return AsyncStorage.setItem(prefix + name, JSON.stringify(value));
	}

	static removeItem(name: string) {
		return AsyncStorage.removeItem(prefix + name);
	}
}
