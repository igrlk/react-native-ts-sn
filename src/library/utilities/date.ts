import { format } from 'date-fns';

export function getDateForSample(miliseconds: number | string) {
  return format(new Date(+miliseconds), 'MMM  d, yyyy');
}
