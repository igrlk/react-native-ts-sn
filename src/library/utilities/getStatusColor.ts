export default function getStatusColor (status: string) {
    switch(status) {
        case 'Sample requested': 
            return { backgroundColor: '#EA4640' }
        case 'Sample Kit Shipped':
            return { backgroundColor: '#FF9A40' }
        case 'sample Received':
            return { backgroundColor: '#0075FF' }
        case 'Processing':
            return { backgroundColor: '#9352DA' }
        case 'Completed':
            return { backgroundColor: '#29D98E' }
        default:
            return { backgroundColor: '#EA4640' }
    }
}
