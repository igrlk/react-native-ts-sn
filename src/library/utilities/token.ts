import AsyncStorage from 'library/utilities/asyncStorage';

import { resetNavigation, navigateTo } from './navigator';

export const name = 'authToken';

export const setToken = (authToken: string) => AsyncStorage.setItem(name, authToken);

export const getToken = async () => {
	const token = await AsyncStorage.getItem(name);
	if (await AsyncStorage.getItem('isTokenForRegistration')) {
		AsyncStorage.removeItem('isTokenForRegistration');
		AsyncStorage.removeItem(name);
	}

	return token;
};

export const checkTokenForLogin = async () => {
	const authToken = await getToken();

	if (authToken) {
		resetNavigation('Landing');

		return true;
	}

	return false;
};

export const removeToken = async () => AsyncStorage.removeItem(name);

export const loginUser = async ({
	authToken,
	email,
	uuid,
	showOnboarding,
}: {
	authToken: string;
	email: string;
	uuid: string;
	showOnboarding?: boolean;
}) => {
	await Promise.all([
		AsyncStorage.setItem(name, authToken),
		AsyncStorage.setItem('email', email),
		AsyncStorage.setItem('uuid', uuid),
	]);
	if (showOnboarding) {
		resetNavigation('Onboarding');
	} else {
		resetNavigation('Landing');
	}
};

export const logout = () => {
	AsyncStorage.removeItem(name);
	resetNavigation('Login');
};
