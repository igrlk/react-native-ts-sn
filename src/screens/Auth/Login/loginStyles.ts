import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	orContainer: {
		marginTop: 12,
		marginBottom: 12,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		position: 'relative',
	},
	orText: {
		color: '#929292',
		fontFamily: 'Poppins',
		fontSize: '$fontSize.mini',
		paddingLeft: 5,
		paddingRight: 5,
		backgroundColor: '$colorWhite',
	},
	orLine: {
		width: '100%',
		height: 1,
		backgroundColor: '#E1E1E1',
		position: 'absolute',
		top: '50%',
	},
	inputEmail: {
		marginTop: 30,
		marginBottom: 25,
	},
	inputPassword: {
		marginBottom: 33,
	},
	forgotPassword: {
		color: '$colorBlue',
		fontSize: '$fontSize.xmini',
		marginBottom: 3,
	},
	logo: {
		marginTop: 59,
		marginBottom: 34,
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
	},
});
