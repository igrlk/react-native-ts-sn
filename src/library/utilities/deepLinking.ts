import { LinkingStatic } from 'react-native';

import { navigateTo } from 'library/utilities/navigator';

export type DeepLinkingUrl = string | { url: string } | null;

export function handleDeepLinking(deepLinkingUrl: DeepLinkingUrl) {
  const url = getUrl(deepLinkingUrl);
  console.log(url);
  if (!url) return;

  if (url.indexOf('signup/') === 0) {
    navigateTo('SignUpAddPassword', getParamsForSignUp(url));
  }
}

export function getUrl(deepLinkingUrl: DeepLinkingUrl) {
  const preprocessedUrl = deepLinkingUrl
    ? typeof deepLinkingUrl === 'string'
      ? deepLinkingUrl
      : deepLinkingUrl.url
    : null;

  if (!preprocessedUrl) return;

  return preprocessedUrl.slice(14);
}

export function getParamsForSignUp(url: string) {
  const [signUpEmail, signUpKey] = url.slice(7).split('/');

  return { signUpKey, signUpEmail };
}

export function bindDeepLinkingHandlers(Linking: LinkingStatic) {
  return () => {
    Linking.getInitialURL().then(handleDeepLinking);
    Linking.addEventListener('url', handleDeepLinking);
  };
}
