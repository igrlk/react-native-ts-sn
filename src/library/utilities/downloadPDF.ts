import RNHTMLtoPDF from 'react-native-html-to-pdf';

import { navigateTo } from 'library/utilities/navigator';

export default function dowloadPdf(options: any, screenName: string, backToScreen: string) {
    RNHTMLtoPDF.convert(options).then((file: any) => {
        navigateTo(screenName, { backToScreen });
    })
    .catch((err: Error) => {});
}