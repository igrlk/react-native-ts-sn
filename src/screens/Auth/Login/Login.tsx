import React, { useEffect, useState } from 'react';
import { View, Text } from 'react-native';

import Screen from 'library/common/commonComponents/Screen';
import WithLoader from 'library/common/commonComponents/WithLoader';
import Button from 'library/common/commonComponents/Buttons/Button';
import Input from 'library/common/commonComponents/Inputs/Input';
import LoginLogo from 'resources/images/svg/LoginLogo';
import { getNavigateTo } from 'library/utilities/navigator';
import useLogin from './loginHooks/useLogin';

import styles from './loginStyles';
import { checkTokenForLogin } from 'library/utilities/token';

export default function Login() {
	const [isCheckingToken, setIsCheckingToken] = useState(true);

	const {
		email,
		setEmail,
		emailError,
		password,
		setPassword,
		passwordError,
		makeLogin,
		loading,
	} = useLogin();

	useEffect(() => {
		checkTokenForLogin().then(r => !r && setIsCheckingToken(false));
	}, []);

	return (
		<Screen>
			<WithLoader loading={isCheckingToken}>
				<View style={styles.logo}>
					<LoginLogo />
				</View>

				<WithLoader loading={loading}>
					<Input
						value={email}
						onChange={setEmail}
						label='Email'
						style={styles.inputEmail}
						error={emailError}
					/>
					<Input
						value={password}
						onChange={setPassword}
						label='Password'
						rightLabel={
							<Text style={styles.forgotPassword} onPress={getNavigateTo('ForgotPasswordCreate')}>
								Forgot password
							</Text>
						}
						style={styles.inputPassword}
						error={passwordError}
					/>

					<Button onClick={makeLogin}>Login</Button>

					<View style={styles.orContainer}>
						<View style={styles.orLine} />
						<Text style={styles.orText}>or</Text>
					</View>

					<Button onClick={getNavigateTo('SignUpCreate')}>Create an Account</Button>
				</WithLoader>
			</WithLoader>
		</Screen>
	);
}
