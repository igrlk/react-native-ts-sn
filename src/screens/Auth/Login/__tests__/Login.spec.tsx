import React from 'react';
import renderer from 'react-test-renderer';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from '@apollo/react-hooks';

import Login from '../Login';
import { getToken } from 'library/utilities/token';

const client = new ApolloClient({
	uri: 'http://sentinel.oioio.ru:8889/graphql',
	fetch: jest.fn(),
	request: async operation => {
		const authToken = await getToken();
		operation.setContext({
			headers: {
				authorization: authToken ? `Bearer ${authToken}` : '',
			},
		});
	},
});

describe('Login', () => {
	it('match to snapshot', () => {
		const tree = renderer
			.create(
				<ApolloProvider client={client}>
					<Login />
				</ApolloProvider>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
