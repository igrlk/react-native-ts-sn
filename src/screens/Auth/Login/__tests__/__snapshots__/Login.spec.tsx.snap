// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <View
            onLayout={[Function]}
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="small"
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
